doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,64 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas: - C programming interface (function prototypes,
+structures, and macros), provided in C header files - Configuration system (Kconfig) - Device driver
+model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- Private
+    These APIs are intended for use within boundary of a :term:`software component`. Private APIs
+    defined in Zephyr tree are not subject to :ref:`api_lifecycle`. Therefore, can be changed or
+    removed at any time. Changes to the private API may not be documented.
+
+- Internal
+    These APIs are intended for use only between certain software components. The context
+    where the API is called or implemented is well defined. For example, functions prefixed with
```

Comment:
I just came across Zephyr naming convention wiki: https://github.com/zephyrproject-rtos/zephyr/wiki/Naming-Conventions

That doc likely needs updating for consistency with this doc.

Reply:
I'd suggest removing (or move to docs) this wiki page altogether.